### ISSUE TITLE
R: Package dependencies

### ISSUE THREAD

**Issue by csgillespie** [travis-ci/travis-ci#5756]
My package is failing to build due to a strange dependency issue (for package `nlme`). The current version on CRAN is 3.1-125, but   at https://travis-ci.org/rcourses/nclRpredictive/builds/114224002#L3301

I get 

```
** preparing package for lazy loading
Error in loadNamespace(j <- i[[1L]], c(lib.loc, .libPaths()), versionCheck = vI[[j]]) : 
  namespace ‘nlme’ 3.1-122 is already loaded, but >= 3.1.123 is required
ERROR: lazy loading failed for package ‘nclRpredictive’
```

However,  this

https://travis-ci.org/rcourses/nclRpredictive/builds/114224002#L676

and

https://travis-ci.org/rcourses/nclRpredictive/builds/114224002#L1581

suggests that `nlme` has been installed.

---
Pinging: @craigcitro, @hadley, @jimhester

**Comment by jimhester** [comment]
This isn't really a travis issue, I get the same error if I clone your package and run `devtools::install_deps("nclRpredictive")`.

I think the reason is because nlme is a recommended package included in base R that older version is already attached and is loaded. A fix which worked on my machine is to explicitly install _just_ nlme first, then install the remaining dependencies. You can do this on Travis by adding a `before_install` step.

```yaml
before_install:
  - Rscript -e 'install.packages("nlme")'
```

**Comment by csgillespie** [comment]
I've tried your suggestion, but no luck. See https://travis-ci.org/rcourses/nclRpredictive#L474

I'm probably missing something, but the problem line is:

```
 R CMD check nclRpredictive_1.1.1.tar.gz --as-cran
```

Surely this is a new R process so should get correct version of nlme from /home/travis/R/library

**Comment by jimhester** [comment]
I agree it looks like it is installing the new version of `nlme` fine, so I am not sure why it isn't being used, I can reproduce the same error locally and setting `R_LIBS_USER` and installing the new nlme package fixes the issue, so it should work on Travis as well.

Perhaps you should try deleting your cache files in https://travis-ci.org/rcourses/nclRpredictive/caches and try again.

**Comment by csgillespie** [comment]
It's not a cache issue, since the package is yet to build so I don't have a cache

```
No caches have been created yet,
read more on how to setup caching with your build.
```

**Comment by jepusto** [comment]
I am having a very similar problem with [my package](https://github.com/jepusto/clubSandwich). It looks like the problem is that R CMD check is using a different set of libraries than where nlme is installed. In [this build](https://travis-ci.org/jepusto/clubSandwich/builds/115843253), nlme is installed into [/home/travis/R/Library/nlme/libs](https://travis-ci.org/jepusto/clubSandwich/builds/115843253#L529). In the testthat.R script, I try looking for nlme [but can't find it](https://travis-ci.org/jepusto/clubSandwich/builds/115843253#L822) because that library path isn't showing up:
` > library(testthat)
  > library(clubSandwich)
  > .libPaths()
  [1] "/tmp/RtmpiKjVVk/RLIBS_21715f55aeab" "/home/travis/R-bin/lib/R/library"  
  > sapply(.libPaths(), packageVersion, pkg="nlme")
  Error in FUN(X[[i]], ...) : package 'nlme' not found
  Calls: sapply -> lapply -> FUN
  Execution halted DONE
`
I'm not really sure how to work around this. I'd be grateful for any suggestions.

**Comment by csgillespie** [comment]
The following (__extremely__ hacky) solution seems to work

````
before_install:
    - sudo rm -fvr /usr/local/lib/R/site-library/nlme
    - sudo rm -fvr /home/travis/R-bin/lib/R/library/nlme
    - Rscript -e 'install.packages("nlme", lib="/home/travis/R-bin/lib/R/library/")'
```

I suspect I can delete one of the `sudo` calls.

**Comment by hadley** [comment]
FWIW the nlme issue appears to be a bug in R

**Comment by jimhester** [comment]
@jepusto your error is simply because you do not have nlme in Suggests, which is required for any packages used in your tests.

In general the only package you should need to use `r_packages` for is covr. Any packages that are used to build or check your package need to be listed in your DESCRIPTION.

**Comment by jepusto** [comment]
@jimhester Actually I do have nlme suggested in [my description file](https://github.com/jepusto/clubSandwich/blob/master/DESCRIPTION). Or am I looking in the wrong place?

**Comment by jimhester** [comment]
@jepusto no you are right sorry.

I think both of your issues can be fixed by simply installing the new nmle version over the old one in `/home/travis/R-bin-lib/R/library/`

```yaml
before_isntall:
  - Rscript -e 'install.packages("nlme", lib="/home/travis/R-bin/lib/R/library/")'
```
See James' successful build at https://travis-ci.org/jepusto/clubSandwich/builds/115867566. No need to delete the existing nlme or use `sudo`.

**Comment by jepusto** [comment]
Thanks for the tips, y'all! My build is running successfully. Now back to unit testing....

**Comment by pitakakariki** [comment]
Is this the same problem as https://github.com/travis-ci/travis-ci/issues/6850?

A recommended package on Ubuntu is just far enough behind CRAN that there's a version problem, and it's not being updated when dependencies are installed?
